pyazhpc/arm.py

```diff
@@ -358,11 +360,16 @@ def __helper_arm_create_datadisks(self, sizes, sku, cache):
         return datadisks
 
     def __helper_arm_create_image_reference(self, refstr):
-        return {
-            "publisher": refstr.split(":")[0],
-            "offer": refstr.split(":")[1],
-            "sku": refstr.split(":")[2],
-            "version": refstr.split(":")[3]
+        if re.search(":", refstr):
+           return {
+              "publisher": refstr.split(":")[0],
+              "offer": refstr.split(":")[1],
+              "sku": refstr.split(":")[2],
+              "version": refstr.split(":")[3]
+        }
+        else:
```

Comment:
Are you sure that you can't use : in a custom image name ? If so this is going to break this, otherwise this is good.